prereise/gather/demanddata/bldg_electrification/floor_area_avg_slopes.py

```diff
@@ -0,0 +1,608 @@
+            )
+            * puma_data_zone["frac_in_zone"]
+        ).sum()
+        total_area_zone_heat_com = (
+            (
+                puma_data_zone[f"com_area_{base_year}_m2"]
+                * puma_data_zone[f"frac_elec_sh_com_{base_year}"]
+            )
+            * puma_data_zone["frac_in_zone"]
+        ).sum()
+        total_area_res = (
+            puma_data_zone[f"res_area_{base_year}_m2"] * puma_data_zone["frac_in_zone"]
+        ).sum()
+        total_area_com = (
+            puma_data_zone[f"com_area_{base_year}_m2"] * puma_data_zone["frac_in_zone"]
+        ).sum()
+        zone_floor_area = pd.concat(
+            [
+                zone_floor_area,
+                pd.DataFrame(
+                    {
+                        "total res": total_area_res,
+                        "total com": total_area_com,
+                        "heat": total_area_zone_heat_res + total_area_zone_heat_com,
+                        "% Heated": (
+                            (total_area_zone_heat_res + total_area_zone_heat_com)
+                            / (total_area_res + total_area_com)
+                        )
+                        * 100,
+                        "res heat": total_area_zone_heat_res,
+                        "com heat": total_area_zone_heat_com,
+                        "cool": total_area_zone_cool_res + total_area_zone_cool_com,
+                        "res cool": total_area_zone_cool_res,
+                        "com cool": total_area_zone_cool_com,
+                    },
+                    index=[zone],
+                ),
+            ]
+        )
+
+    zone_floor_area.to_csv(
+        f"./Profiles/result_stats/{iso_name[iso]}_zone_floor_area_m2.csv"
+    )
+    return zone_floor_area
+
+
+def state_shp_overlay(state):
+    """Select load zones within a state
+
+    :param str state: abbrev. of state
+    :return: (*geopandas.GeoDataFrame*) state boundaries and load zones within it
+    """
+
+    state_shp = state_shapefile[state_shapefile["STUSPS"] == state]
+
+    zone_shp = gpd.GeoDataFrame(
+        gpd.read_file(
+            os.path.join(os.path.dirname(__file__), "shapefiles", "ba_area.shp")
+        )
+    ).to_crs("EPSG:4269")
+
+    zone_shp["area"] = zone_shp["geometry"].to_crs({"proj": "cea"}).area
+
+    zone_state = gpd.overlay(zone_shp, state_shp.to_crs("EPSG:4269"))
+    zone_state["area"] = zone_state["geometry"].to_crs({"proj": "cea"}).area
+    zone_state["area_frac"] = [
+        zone_state["area"][i]
+        / list(zone_shp[zone_shp["BA"] == zone_state["BA"][i]]["area"])[0]
+        for i in range(len(zone_state))
+    ]
+    zone_state.loc[zone_state["area_frac"] >= 0.99, "area_frac"] = 1
+    zone_state = zone_state.drop(zone_state[zone_state["area_frac"] <= 0.00001].index)
+
+    return zone_state
+
+
+def main_plots(iso, plot_show):
+    """Creats floor area avraged slopes for all zones within the ISO for one year.
+
+    :param str iso: abbrev. name of ISO.
+
```

Comment:
Missing doc string for `plot_show` here, i.e. `:param bool plot_show: show the plot or not, default to True.` We can also make it a keyword argument instead, i.e. `def main_plots(iso, plot_show=True)`

Reply:
Made the changes

Reply:
Made the changes